server/src/apub/shared_inbox.rs

```diff
@@ -0,0 +1,222 @@
+use super::*;
+
+#[serde(untagged)]
+#[derive(Serialize, Deserialize, Debug)]
+pub enum SharedAcceptedObjects {
+  Create(Create),
+  Update(Update),
+}
+
+impl SharedAcceptedObjects {
+  fn object(&self) -> Option<&BaseBox> {
+    match self {
+      SharedAcceptedObjects::Create(c) => c.create_props.get_object_base_box(),
+      SharedAcceptedObjects::Update(u) => u.update_props.get_object_base_box(),
+    }
+  }
+}
+
+/// Handler for all incoming activities to user inboxes.
+pub async fn shared_inbox(
+  request: HttpRequest,
+  input: web::Json<SharedAcceptedObjects>,
+  db: DbPoolParam,
+  chat_server: ChatServerParam,
+) -> Result<HttpResponse, Error> {
+  // TODO: would be nice if we could do the signature check here, but we cant access the actor property
+  let activity = input.into_inner();
+  let conn = &db.get().unwrap();
+
+  let json = serde_json::to_string(&activity)?;
+  debug!("Shared inbox received activity: {}", json);
+
+  let object = activity.object().cloned().unwrap();
+
+  match (activity, object.kind()) {
+    (SharedAcceptedObjects::Create(c), Some("Note")) => {
```

Comment:
Would be nice to use NoteType.to_string() instead of a raw string, but I couldnt get that working.